domain/src/main/java/com/nextstep/edu/domain/CalculationExpression.kt

```diff
@@ -1,18 +1,66 @@
 package com.nextstep.edu.domain
 
-class CalculationExpression {
-    fun split(inputValue: String?): List<String> {
-        require(!inputValue.isNullOrBlank()) { "입력 값이 Null입니다." }
-        return inputValue.split(INPUT_VALUE_DELIMITER)
+
+class CalculationExpression(private var value: List<Any> = emptyList()) {
```

Comment:
```suggestion
class CalculationExpression(private val value: List<Any> = emptyList()) {
```

value 객체를 직접적으로 새로 선언하는 케이스가 없는 것 같아요~ 이런 경우 val로 변경해주는 것이 좋습니다.
그리고 테스트코드에서 주입을 위해 생성자에 같이 받았지만 캡슐화를 위해 private으로 선언한 것으로 보이는데요 👍 👍 

이런 경우 외부에서 읽을 value를 위해 getValue() 함수를 만들지 않고
`val expression: List<Any> get() = value.toList()` 와 같은 형식으로 진행해주는 것이 더욱 좋습니다.
프로퍼티는 getter와 setter를 기본적으로 가지고있기 때문에 프로퍼티를 이용하는 것이 조금 더 코틀린스러운 코드라고 생각되네요!

추가적으로 value는 List 타입으로 되어있고, 생성자에서 주입받다보니 다양한 리스트 타입을 받을 수 있는 List타입으로 결정했다는 생각이 드네요! (맞나요? 😅😅)
Collection.plus 연산의 경우 ArrayList 객체를 새로 만들어서 List를 갈아끼우는 형태를 하고 있어요.
따로 MutableList 타입으로 프로퍼티를 가지게 하는 것도 생각해보시면 좋을 것 같아요 (이 부분은 장단점, 취향적인 이유가 있기 때문에 반영하지 않으셔도 됩니다! 추가적으로 다른 생각 있으시면 코멘트 남겨주세요! 저도 작성하면서 어떤게 더 좋은지 계속 고민하게 되네요🧐🧐)

Reply:
val로 선언한 이유는 말씀해주신 것처럼 ArrayList 객체를 새로 만들어서 List를 갈아끼우는 형태라서 아래와 같은 경우 var일 때만 재할당이 되기 때문입니다. 최대한 MutableList를 안쓰려고 하다보니 이렇게 되었네요. 😢 

`value.isNullOrEmpty() -> value += operand`

Reply:
MutableList를 안쓰는 것보단 상황에 맞춰 잘 사용하는 것이 중요하다고 생각해요!

Mutable한 리스트를 잘못 사용하는 경우 의도와는 다른 변경이 일어날 수도 있습니다. 
하지만 리스트를 수정하는 책임을 CalculationExpression가 잘 갖게되고 다른 객체들이 책임을 나눠갖지 않게된다면 오사용의 케이스가 거의 생기지 않게 됩니다.

외부에서는 read-only로 리스트에 접근하기 위해서 backing property를 사용하는 것이 깔끔하다고 생각해요!

요점은 객체에 책임 분리를 잘 한다면 MutableList를 사용해도 크게 이슈가 발생하지 않습니다!

Reply:
> MutableList 타입으로 프로퍼티를 가지게 하는 것도 생각해보시면 좋을 것 같아요

이 부분이 오해의 소지가 있을까봐 추가 코멘트 남기면..! 생성자에서 List로 받는 것은 유지하고, 객체 안에서 수정될 멤버 프로퍼티를 추가로 선언하자였습니다. 
생성자에서 List가 제일 추상화된 형태이기 때문에 List로 받는 것은 수정하면 안됩니다..! (이 부분이 살짝 애매하기도하고.. 더 좋은 방법이 있을 것 같은데 따로 생각이 나지 않아서 수정하지 않으셔도 된다라고 남겼습니다..ㅎㅎ;)